### PR TITLE
feat: MC2010 Creep and shrinkage

### PR DESCRIPTION
**Pull request by slobbeat**

### CODE REVIEW

**Comment by mortenengen** [comment]
I am sorry. I accidentally closed the PR when I deleted the branch in #7...

**Comment by slobbeat** [comment]
> Hi @slobbeat 😊 Could you please merge the latest changes from dev into your branch?

Hi @mortenengen , maybe I did something wrong, but my [last commit](https://github.com/fib-international/structuralcodes/pull/11/commits/b8adf0d639dd785c7e562be4aa7f5fd9b4d51594) was meant for this. If I compare the recent changes in dev with this branch, they are all there. Please, let me know if you think this is not the case, and I will have another look.

**Comment by mortenengen** [comment]
@slobbeat, I added some minor changes to comply with flake8 and pylint. @talledodiego will do a review :smile:

File: structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py
Comment (talledodiego) [inline structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py:625]: same comment as before if time is for sure a numpy array or can be a float. The output can change accordingly based on user input?

File: structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py
Comment (talledodiego) [inline structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py:655]: same as before about array

File: structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py
Comment (talledodiego) [inline structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py:60]: Raises a warning instead of print a warning

File: structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py
Comment (talledodiego) [inline structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py:691]: should we explicitly state that fcm is fcm(t0), so at time t0?

File: structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py
Comment (talledodiego) [inline structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py:701]: the only inputs parameters should be phi_bc and phi_dc. We already have functions to compute them.
The body of the functions simplifies greatly.
The logic of this current function (as in previous cases) is more related to the concreteMC2010 class.

File: structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py
Comment (talledodiego) [inline structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py:776]: The signature should have only Eci(t0), Eci and phi(t,t0). We have the functions to compute them.
As before the logic of this function can be used for the concreteMC2010 class? Or we could think about a further function here that uses this logic That is additional to the small equation that computes the second term of 5.1-61

**Review by talledodiego — CHANGES_REQUESTED** [review]
@slobbeat I took a look to the code. Thanks for your excellent job. 
In general I would tend to be more coherent to the equations in the code in the methods signatures.
You can find my comments in this review.
I did not go through test functions waiting for the changes suggested in the reviews.

File: structuralcodes/codes/mc2010/_concrete_material_properties.py
Comment (mortenengen) [reply structuralcodes/codes/mc2010/_concrete_material_properties.py:247]: Good point, but in the case of establishing a creep compliance matrix it would be useful to be able to work on arrays instead of element by element. `np.sqrt` accepts `ArrayLike`, so this could be a step in that direction 😃

**Comment by slobbeat** [comment]
@talledodiego @mortenengen : thanks for the review. Your comments has been processed, so maybe you can have another look to it?

File: structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py
Comment (talledodiego) [inline structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py:218]: Maybe this portion simplifies and improves performance if we use the method from numpy `asarray`? So we don't need to check the length before. It converts anything that can be expressed as an array to your values. Positive side note: if we already pass a numpy array, no copy is performed.
For typing, shall we stick with float|list or it is better to use ArrayLike? What do you say @mortenengen ?

File: structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py
Comment (talledodiego) [inline structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py:278]: Maybe we should use ArrayLike? so that the user can input an array, or a list, or anything that can be converted to array?

File: structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py
Comment (talledodiego) [inline structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py:389]: same comment as before regarding ArrayLike?

File: structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py
Comment (talledodiego) [inline structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py:441]: Sme comment as before for ArrayLike?

File: structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py
Comment (talledodiego) [inline structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py:597]: same comment as before for ArrayLike?

File: structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py
Comment (talledodiego) [inline structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py:674]: I don't know here if the user is expected that this function is 5.1-63 it is strange to provide sigma and fcm. Those are used for effect of high stresses in 5.1-74.
I would split the phi function for 5.1-63 and phi_sigma for 5.1-74.
The signatures would then become:
`def phi(_phi_bc, _phi_dc)`
and
`def phi_sigma(phi, sima, fcm)`

I have not strong feelings in doing so, but if we want to leave as it is it should be updated the docstrings highlighting that this function is implementation of5%.1-63 + 5.1-74?

**Review by talledodiego — CHANGES_REQUESTED** [review]
@slobbeat thanks for your great work in making the changes.
I am putting some minor comments that you can see if you feel worth accepting.
I resolved all comments that have been addressed in this revision.

After that it is ok for merging from me.

File: structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py
Comment (mortenengen) [reply structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py:218]: Good if we can stick to `ArrayLike` typing in order to be consistent, and great if we can avoid unecessary copies by using `.asarray()` 😃

File: structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py
Comment (mortenengen) [reply structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py:674]: Good comment. I think we can always refactor later if it is needed, but the docstring should always be clear. @slobbeat, I suggest to keep the function as is and just adjust the docstring 😃

File: structuralcodes/codes/mc2010/_concrete_material_properties.py
Comment (slobbeat) [reply structuralcodes/codes/mc2010/_concrete_material_properties.py]: This is a property that is related to the development of the concrete over time. It is, however, not only limited to creep and shrinkage.

File: structuralcodes/codes/mc2010/_concrete_material_properties.py
Comment (slobbeat) [reply structuralcodes/codes/mc2010/_concrete_material_properties.py]: This is a property that is related to the development of the concrete over time. It is, however, not only limited to creep and shrinkage.

File: structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py
Comment (slobbeat) [reply structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py:57]: The Model Code indicates that the model is valid for ordinary concrete (20 ≤ fcm ≤ 130 MPa), suggesting that it is indeed the 28 day strength. For the compressive stress a distinction is made related to the compressive strength at time of loading: |σc| ≤ 0.4fcm(t0).

File: structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py
Comment (slobbeat) [reply structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py]: Done

File: structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py
Comment (slobbeat) [reply structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py]: Done

File: structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py
Comment (slobbeat) [reply structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py]: Done

File: structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py
Comment (slobbeat) [reply structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py]: Done

File: structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py
Comment (slobbeat) [reply structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py]: Done

File: structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py
Comment (slobbeat) [reply structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py:673]: Done

**Comment by slobbeat** [comment]
Credits goes to Gijs, who actually made the changes!

**Review by mortenengen — APPROVED** [review]

**Review by talledodiego — APPROVED** [review]
Excellet. Very good job

**Pull request merged**
